src/lib.rs

```diff
@@ -166,14 +166,25 @@ impl Volume for S3Volume {
             .await,
         );
 
-        client
-            .create_bucket(config.reuse_bucket_is_enabled)
-            .await
-            .map_err(|e| zerror!("Couldn't create storage: {e}"))?
-            .map_or_else(
-                || tracing::debug!("Reusing existing bucket '{}'.", client),
-                |_| tracing::debug!("Bucket '{}' successfully created.", client),
-            );
+        // In the crate aws-sdk-s3, it tries to get the current runtime, but it will fail if we load the backend as a dynamic plugin.
+        // In this case, we need to provide our own runtime to avoid the issue.
+        match tokio::runtime::Handle::try_current() {
+            Ok(_) => client.create_bucket(config.reuse_bucket_is_enabled).await,
+            Err(_) => {
+                let c_client = client.clone();
+                TOKIO_RUNTIME
+                    .spawn(
+                        async move { c_client.create_bucket(config.reuse_bucket_is_enabled).await },
+                    )
+                    .await
+                    .map_err(|e| zerror!("Unable to spawn the task for create_bucket: {e}"))?
```

Comment:
Given that this code is used multiple times it may make sense to have a function that takes the future and does the whole match thing.
What do you think?